scene/2d/camera_2d.cpp

```diff
@@ -409,6 +409,35 @@ void Camera2D::force_update_scroll() {
 	_update_scroll();
 }
 
+void Camera2D::reset_smoothing() {
+
+	smoothed_camera_pos = camera_pos;
+	_update_scroll();
+}
+
+void Camera2D::align() {
+
+	Size2 screen_size = get_viewport_rect().size;
+	screen_size=get_viewport_rect().size;
+	Point2 current_camera_pos = get_global_transform().get_origin();
+	if (anchor_mode==ANCHOR_MODE_DRAG_CENTER) {
+		if (h_ofs<0) {
```

Comment:
[cam2d-test.tar.gz](https://github.com/godotengine/godot/files/341849/cam2d-test.tar.gz)
This is an example project that I used for testing.